File: src/UrlShortener/Contracts/V1/ApiRoutes.cs
﻿namespace UrlShortener.Contracts.V1;

public static class ApiRoutes
{
    private const string Root = "api";
    private const string Version = "v{version:apiVersion}";
    private const string Base = Root + "/" + Version + "/";

    public static class Account
    {
        private const string Endpoint = Base + "account";

        public const string Login = Endpoint + "/login";

        public const string Register = Endpoint + "/register";
    }

    public static class Config
    {
        private const string Endpoint = Base + "config";

        public const string Get = Endpoint + "/get";
    }

    public static class CustomUrl
    {
        private const string Endpoint = Base + "customurl";

        public const string Get = Endpoint + "/get/{shortUrl}";

        public const string Save = Endpoint + "/save";
    }

    public static class Url
    {
        private const string Endpoint = Base + "url";

        public const string Get = Endpoint + "/get/{shortUrl}";

        public const string Save = Endpoint + "/save";
    }
}

File: src/UrlShortener/models/apiRoutes.ts
export class ApiRoutes {
  static readonly root = 'https://localhost:7081'
  static readonly version = '1'
  static readonly baseUrl = `${this.root}/v${this.version}`

  static Account = class {
    private static readonly endpoint = `${ApiRoutes.baseUrl}/account`

    public static readonly Login = `${this.endpoint}/login`
    public static readonly Register = `${this.endpoint}/register`

    static Email = class {
      private static readonly endpoint = `${ApiRoutes.Account.endpoint}/email`

      public static readonly Confirm = `${this.endpoint}/confirm`
      public static readonly IsConfirmed = `${this.endpoint}/isConfirmed`
      public static readonly ResendVerification = `${this.endpoint}/resendVerification`
    }

    static Password = class {
      private static readonly endpoint = `${ApiRoutes.Account.endpoint}/password`

      public static readonly VerifyToken = `${this.endpoint}/verifyToken`
      public static readonly Reset = `${this.endpoint}/reset`
      public static readonly SendRecovery = `${this.endpoint}/recovery`

      public static readonly Change = `${this.endpoint}/change`
    }
  }
}
